src/toil/fileStores/nonCachingFileStore.py

```diff
@@ -267,21 +270,27 @@ def _createJobStateFile(self) -> str:
         Create the job state file for the current job and fill in the required
         values.
 
+        Places the file in the coordination directory.
+
         :return: Path to the job state file
         :rtype: str
         """
-        jobStateFile = os.path.join(self.localTempDir, '.jobState')
-        jobState = {'jobProcessName': get_process_name(self.workDir),
+        jobState = {'jobProcessName': get_process_name(self.coordination_dir),
                     'jobName': self.jobName,
                     'jobDir': self.localTempDir}
-        with open(jobStateFile + '.tmp', 'wb') as fH:
+        (fd, jobStateFile) = tempfile.mkstemp(suffix='.jobState.tmp', dir=self.coordination_dir)
+        with open(fd, 'wb') as fH:
+            # Write data
             dill.dump(jobState, fH)
+        # Drop suffix
+        jobStateFile = jobStateFile[:-4]
```

Comment:
Nit: For clarity, consider `jobStateFile[:-len('.tmp')]`.

Reply:
```suggestion
        jobStateFile = jobStateFile[:-len('.tmp')]
```